spring-kafka/src/main/java/org/springframework/kafka/listener/BatchErrorHandler.java

```diff
@@ -42,4 +44,20 @@ default void handle(Exception thrownException, ConsumerRecords<?, ?> data, Consu
 		handle(thrownException, data);
 	}
 
+	/**
+	 * Handle the exception.
+	 * @param thrownException the exception.
+	 * @param data the consumer records.
+	 * @param consumer the consumer.
+	 * @param container the container.
+	 * @param invokeListener a callback to re-invoke the listener.
+	 * @throws InterruptedException if the thread is interrupted.
+	 * @since 2.3.7
+	 */
+	default void handle(Exception thrownException, ConsumerRecords<?, ?> data,
+			Consumer<?, ?> consumer, MessageListenerContainer container, Supplier<Boolean> invokeListener) {
```

Comment:
Shouldn't this one be without a `container`?
Why do we need then that `ContainerAwareBatchErrorHandler` otherwise?

Reply:
Notice how each sub interface provides a default implementation of the super; this avoids implementations from having to implement other methods.

Adding the method here avoids `instanceof` checks in the container; in fact, I now realize I can get rid of the one `instanceof` check that is there now.